src/lib.rs

```diff
@@ -886,11 +904,104 @@ fn membership_in_clone(pkg: &Package, repo_dir: &Path) -> Result<bool> {
+        if is_same_package_except_name(pkg, &table) {
+            return Ok(true);
+        }
     }
 
     Ok(false)
 }
 
+/// Checks if a package is the same as one defined in a Cargo.toml table, except for its name.
+/// This helps identify renamed packages.
+fn is_same_package_except_name(pkg: &Package, cargo_toml: &Table) -> bool {
+    let Some(pkg_table) = cargo_toml.get("package").and_then(Value::as_table) else {
+        return false;
+    };
+
+    // Check repository URL (if present in both)
+    if let (Some(original_repo), Some(candidate_repo)) = (
+        &pkg.repository,
+        pkg_table.get("repository").and_then(Value::as_str),
+    ) {
+        if original_repo == candidate_repo {
+            return true;
+        }
+    }
+
+    // Check other invariant fields
+    // 1. Check authors (if present in both)
+    if !pkg.authors.is_empty() {
+        if let Some(candidate_authors) = pkg_table.get("authors").and_then(Value::as_array) {
+            let candidate_authors: Vec<&str> = candidate_authors
+                .iter()
+                .filter_map(|a| a.as_str())
+                .collect();
+
+            if !candidate_authors.is_empty() && have_common_author(&pkg.authors, &candidate_authors)
+            {
+                return true;
+            }
+        }
+    }
+
+    // 2. Check version (exact match)
+    if let Some(version_str) = pkg_table.get("version").and_then(Value::as_str) {
+        if let Ok(version) = Version::from_str(version_str) {
+            if version == pkg.version {
+                return true;
+            }
+        }
+    }
+
+    // 3. Check description similarity (if present in both)
+    if let (Some(original_desc), Some(candidate_desc)) = (
+        &pkg.description,
+        pkg_table.get("description").and_then(Value::as_str),
+    ) {
+        if high_similarity(original_desc, candidate_desc) {
+            return true;
+        }
+    }
+
+    false
+}
+
+/// Checks if two lists of authors have at least one author in common.
+fn have_common_author(authors1: &[String], authors2: &[&str]) -> bool {
+    for author1 in authors1 {
+        if authors2.contains(&author1.as_str()) {
+            return true;
+        }
+    }
+    false
+}
+
+/// Checks if two strings have high textual similarity.
+/// Returns true if they share a significant portion of words.
+fn high_similarity(s1: &str, s2: &str) -> bool {
+    let s1_words: HashSet<&str> = s1.split_whitespace().collect();
+    let s2_words: HashSet<&str> = s2.split_whitespace().collect();
+
+    if s1_words.is_empty() || s2_words.is_empty() {
+        return false;
+    }
+
+    let common_words = s1_words.intersection(&s2_words).count();
+    let min_words = s1_words.len().min(s2_words.len());
+
+    // Avoid precision loss by doing integer division first
+    let similarity = if min_words > 0 {
+        (common_words * SIMILARITY_SCALE) / min_words
+    } else {
+        0
+    };
+
+    similarity > SIMILARITY_THRESHOLD
+}
```

Comment:
I know you put a lot of work into this. Thank you for that. But let's please just do exact comparisons for now.